src/main/java/com/waverley/WaitCondition.java

```diff
@@ -0,0 +1,30 @@
+package com.waverley;
+
+import lombok.Getter;
+import org.openqa.selenium.By;
+import org.openqa.selenium.support.ui.ExpectedCondition;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+@Getter
+public enum WaitCondition {
+    visible(ExpectedConditions::visibilityOfElementLocated),
+    present(ExpectedConditions::presenceOfElementLocated),
+    //enabled(ExpectedConditions::elementToBeClickable),
+    allvisible(ExpectedConditions::visibilityOfAllElementsLocatedBy),
+    textpresent(ExpectedConditions::textToBePresentInElementLocated),
+    textpresentinvalue(ExpectedConditions::textToBePresentInElementValue);
+    private BiFunction<By, String, ExpectedCondition<?>> type;
```

Comment:
separate enum keys and instance variable with a new line to increase readability of your code